internal/pkg/service/cli/cmd/remote/create/table/cmd.go

```diff
@@ -8,11 +8,21 @@ import (
 
 	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/dependencies"
 	"github.com/keboola/keboola-as-code/internal/pkg/service/cli/helpmsg"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/configmap"
 	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
 	"github.com/keboola/keboola-as-code/pkg/lib/operation/project/remote/create/table"
 )
 
-func TableCommand(p dependencies.Provider) *cobra.Command {
+type Flags struct {
+	StorageAPIHost configmap.Value[string] `configKey:"storage-api-host" configShorthand:"H" configUsage:"if command is run outside the project directory"`
+	Bucket         configmap.Value[string] `configKey:"bucket" configUsage:"bucket ID (required if the tableId argument is empty)"`
+	Name           configmap.Value[string] `configKey:"name" configUsage:"name of the table (required if the tableId argument is empty)"`
+	Columns        configmap.Value[string] `configKey:"columns" configUsage:"comma-separated list of column names"`
+	PrimaryKey     configmap.Value[string] `configKey:"primary-key" configUsage:"columns used as primary key, comma-separated"`
+	ColumnsFrom    configmap.Value[string] `configKey:"columns-from" configUsage:"the path to the table definition file in json"`
+}
+
+func Command(p dependencies.Provider) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "table [table]",
```

Comment:
I suggest to create `DefaultFlags` function in each command, even if it will be only `return Flags{}`, so we don't have to explain to other people, why it's somewhere present and somewhere not.